File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0, sum = 0, obj ={};
    tangerine.forEach((value) => (obj[value] = 0));
    for(let i=0; i<tangerine.length; i++){
        obj[tangerine[i]] +=1;
    }
    obj = Object.values(obj).sort((a, b) => b - a);
    for(let i=0; i<obj.length; i++){
        answer++;
        sum+=obj[i];
        if(sum>=k){
            return answer;
        }
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0, sum = 0;
        HashMap<Integer, Integer> obj = new HashMap<>();
        for(int i : tangerine){
            obj.put(i, obj.getOrDefault(i, 0)+1);
        }
        
        List<Integer> keySet = new ArrayList<>(obj.keySet());
        keySet.sort((a, b)->obj.get(b)-obj.get(a));
        for(Integer integer : keySet){
            answer++;
            sum += obj.get(integer);
            if(sum>=k) break;
        }
        return answer;
    }
}